internal/migrations/migrator.go

```diff
@@ -342,52 +309,57 @@ func (m *Migrator) generateGenesisConfig(ctx context.Context, cfgParams *common.
 	}
 }
 
-func (m *Migrator) PersistLastChangesetHeight(ctx context.Context, tx sql.Executor) error {
+func (m *Migrator) SetMigrationStatus(status types.MigrationStatus) {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-
-	return setLastStoredChangeset(ctx, tx, m.lastChangeset)
+	m.migrationStatus = status
 }
 
-func (m *Migrator) MigrationStatus() types.MigrationStatus {
-	m.mu.RLock()
-	defer m.mu.RUnlock()
-
-	if m.activeMigration == nil {
-		return types.NoActiveMigration
-	}
-
-	if m.halted {
-		return types.MigrationCompleted
-	}
-
-	if m.inProgress {
-		return types.MigrationInProgress
-	}
+func (m *Migrator) PersistLastChangesetHeight(ctx context.Context, tx sql.Executor) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
 
-	return types.MigrationNotStarted
+	return setLastStoredChangeset(ctx, tx, m.lastChangeset)
 }
 
 // GetMigrationMetadata gets the metadata for the genesis snapshot,
 // as well as the available changesets.
-func (m *Migrator) GetMigrationMetadata() (*types.MigrationMetadata, error) {
+func (m *Migrator) GetMigrationMetadata(ctx context.Context) (*types.MigrationMetadata, error) {
 	m.mu.RLock()
 	defer m.mu.RUnlock()
 
-	// if there is no planned migration, return
-	if m.activeMigration == nil {
+	if m.migrationStatus == types.GenesisMigration {
 		return &types.MigrationMetadata{
 			MigrationState: types.MigrationState{
-				Status: types.NoActiveMigration,
+				Status:      types.GenesisMigration,
+				StartHeight: m.genesisMigrationParams.StartHeight,
+				EndHeight:   m.genesisMigrationParams.EndHeight,
 			},
 		}, nil
 	}
 
+	// if there is no planned migration, return
+	if m.migrationStatus == types.NoActiveMigration {
+		metadata := &types.MigrationMetadata{
+			MigrationState: types.MigrationState{
+				Status: types.NoActiveMigration,
+			},
```

Comment:
The three height fields are prime candidates for `omitempty` IMO.  I'm starting a migration on staging again shortly, but when I use these commands I expect I will see zeros where they have no meaning.
Can hold off on any more changes though until I've gone through it again.  Will being testing with this PR stacked on top of @brennanjl's https://github.com/kwilteam/kwil-db/pull/1036